mariadb_kernel/client_config.py

```diff
@@ -14,11 +14,20 @@ def __init__(self, log, name="mariadb_config.json"):
         self.default_config = {
             "user": "root",
             "host": "localhost",
+            "socket": "/tmp/mysqld.sock",
             "port": "3306",
             "password": "",
             "start_server": "True",
             "client_bin": "mysql",
             "server_bin": "mysqld",
+            "db_init_bin": "mysql_install_db",
+            "extra_server_config": ["--pid-file=/tmp/mysqld.pid",
+                                    f"--datadir={os.path.join(os.environ.get('HOME', '/home/jovyan/'), 'work', '.db')}",
+                                    "--skip_log_error"],
+            "extra_db_init_config": [f"--user={os.environ.get('NB_USER', 'jovyan')}",
+                                     "--auth-root-authentication-method=normal",
```

Comment:
Actually I think it would be better not use socket auth and only use "normal" auth (no need to complicate things).
However in my testing I could never get password auth to work (the client got auth denied, is a empy password on root not allowed?)

Reply:
I totally agree that we should have normal password auth by default, but I like a lot that after your PR here people will be able to configure the kernel to connect via socket auth.
Regarding empty password for root. It is allowed for sure. But please make sure that in MariaDB 10.4+, socket authentication is the default, and if you want empty password connection, you need to either login via socket auth in root account (with regular MariaDB client) and then do `set password=password('');` or just create the datadir with the script and add the `--auth-root-authentication-method=normal` option.

Reply:
We have tried all we can think of but I cannot get normal authentication to work (socket auth works fine).

Reply:
If you want, we can get into more details to try to debug the issue, but it's not relevant for this PR in my opinion, adding socket auth option for the user is a great idea and something I wanted to implement too. As long as `--auth-root-authentication-method=normal` is listed as default for `mysql_install_db` in the kernel, I am very happy with the changes.